src/ShopPaymentTransactionBundle/objects/TPkgShopPaymentTransactionManagerEndPoint.class.php

```diff
@@ -223,6 +225,60 @@ protected function updateOrderPaidStatusBasedOnCurrentBalance()
         return $bOrderPaidStatusChanged;
     }
 
+    private function updateRealUsedVoucherValueBasedOnTransactions(TdbPkgShopPaymentTransaction $transaction): void
+    {
+        $oTransactionData = $this->getTransactionDataFromOrder(
+            TPkgShopPaymentTransactionData::TYPE_PAYMENT,
+            null,
+            true
+        );
+        $valueToCapture = $oTransactionData->getTotalValueForItemType(TPkgShopPaymentTransactionItemData::TYPE_PRODUCT);
+
+        // There are products no captured yet. So no voucher handling is needed because not captured products can use open/refunded voucher values.
```

Comment:
```suggestion
        // There are uncaptured products, so no voucher handling is needed because uncaptured products can use open/refunded voucher values.
```

Is this really always the case?

Reply:
yes if the order is still open we cannot reduce the voucher value. On the capture of the last item this was checked again. And if the order has unused voucher value it was remove from the used voucher.